lib/rules/prop-types.js

```diff
@@ -482,14 +482,25 @@ module.exports = {
           }
           return true;
         case 'ObjectTypeAnnotation':
+          let containsObjectTypeSpread = false;
           var shapeTypeDefinition = {
             type: 'shape',
             children: {}
           };
           iterateProperties(annotation.properties, function(childKey, childValue) {
-            shapeTypeDefinition.children[childKey] = buildTypeAnnotationDeclarationTypes(childValue);
+            if (!childKey && !childValue) {
+              containsObjectTypeSpread = true;
+            } else {
+              shapeTypeDefinition.children[ childKey ] = buildTypeAnnotationDeclarationTypes(childValue);
```

Comment:
Indeed, it should, but since that's the style used a few lines down in the same file, I'd hope it'd be apparent :-)